src/scripts.js

```diff
@@ -130,39 +129,46 @@ function postNewData() {
     "minutesActive" : minutesActiveEntry.value,
     "flightsOfStairs" : flightsOfStairsEntry.value
   }
-  let newData = catchData(newEntry);
+
   const date = setNewDate();
   let userSleepData = {
     "userID" : currentUser.id,
     "date" : date,
-    "hoursSlept" : newData.hoursSlept,
-    "sleepQuality" : newData.sleepQuality
+    "hoursSlept" : newEntry.hoursSlept,
+    "sleepQuality" : newEntry.sleepQuality
   }
 
   let userHydrationData = {
     "userID" : currentUser.id,
     "date" : date,
-    "numOunces": newData.numOunces
+    "numOunces": newEntry.numOunces
   }
 
   let userActivityData = {
     "userID" : currentUser.id,
     "date" : date,
-    "numSteps" : newData.numSteps,
-    "minutesActive" : newData.minutesActive,
-    "flightsOfStairs" : newData.flightsOfStairs
+    "numSteps" : newEntry.numSteps,
+    "minutesActive" : newEntry.minutesActive,
+    "flightsOfStairs" : newEntry.flightsOfStairs
   }
 
+  checkUserData(newEntry, userSleepData, userHydrationData, userActivityData);
+}
+
+function checkUserData(newEntry,userSleepData, userHydrationData, userActivityData) {
   let updatingDisplay = document.getElementById('updatingDisplay');
-  postAllUserData(userSleepData, userHydrationData, userActivityData)
-  .then(response => {
-    updatingDisplay.innerHTML = "Updating Your Account...";
-    setTimeout(() => {updatingDisplay.innerHTML = ""}, 1500)
-    console.log(response)
-  })
-  .catch(err => {
-    alert(err.message)
-  })
+  if (!newEntry.hoursSlept || !newEntry.sleepQuality || !newEntry.numOunces || !newEntry.numSteps || !newEntry.minutesActive || !newEntry.flightsOfStairs) {
+    updatingDisplay.innerText = "Please make sure fields are all filled."
+  } else {
```

Comment:
For future iterations we could do error handling so that user can't put in anything but numbers